src/buildtool.ts

```diff
@@ -1,13 +1,18 @@
-// Copyright 2022 The MathWorks, Inc.
+// Copyright 2022-2024 The MathWorks, Inc.
 
 export interface RunBuildOptions {
     Tasks?: string;
+    BuildOptions?: string;
 }
 
 export function generateCommand(options: RunBuildOptions): string {
     let command: string = "buildtool";
     if (options.Tasks) {
         command = command + " " + options.Tasks;
     }
-    return command
+    if (options.BuildOptions) {
+        command = command + " " + options.BuildOptions + " -ignoreUnknownOptions";
```

Comment:
I don't think we need to append `-ignoreUnknownOptions` since the user is providing all the options. `-ignoreUnknownOptions` seems useful if we are providing dedicated checkboxes/text fields for features we are not sure the user's MATLAB version supports. Without those dedicated fields, I feel like we can leave it up to the user to select the options that are appropriate for their MATLAB version. There's also the question about spitting out an undocumented option in logs.

Reply:
If the user is doing a matrix build across a few different releases and wants to use options that were added in only a subset of those releases, would you just envision them adding `-ignoreUnknownOptions` themselves? Also on a sidenote do we document `-ignoreUnknownOptions`?

Reply:
I think `-ignoreUnknownOptions` would sound awkward as a documented option!

Reply:
I would envision them making a matrix variable with the options that are specific to that version of MATLAB.

Or, if we think that is too much trouble, we might consider stealthily adding `-ignoreUnknownOptions` but I don't think it should be spit out in the logs. It also introduces the potential for mistyped options to go unnoticed.

Reply:
Good point. I think it would be a slight level of hassle for users to provide `build-options` as a matrix variable but I can't think of a way to stealthily add `-ignoreUnknownOptions` to the buildtool command. Do you have anything in mind?